docs/database/includes/azure-postgresql-client.md

```diff
@@ -2,42 +2,167 @@
 ms.topic: include
 ---
 
-### Add Azure authenticated Npgsql client
-
-By default, when you call `AddAzurePostgresFlexibleServer` in your PostgreSQL hosting integration, it configures [📦 Azure.Identity](https://www.nuget.org/packages/Azure.Identity) NuGet package to enable authentication:
+To get started with the .NET Aspire Azure PostgreSQL client integration, install the [📦 Aspire.Azure.Npgsql](https://www.nuget.org/packages/) NuGet package in the client-consuming project, that is, the project for the application that uses the PostgreSQL client. The PostgreSQL client integration registers an [NpgsqlDataSource](https://www.npgsql.org/doc/api/Npgsql.NpgsqlDataSource.html) instance that you can use to interact with PostgreSQL.
 
 ### [.NET CLI](#tab/dotnet-cli)
 
 ```dotnetcli
-dotnet add package Azure.Identity
+dotnet add package Aspire.Azure.Npgsql
 ```
 
 ### [PackageReference](#tab/package-reference)
 
 ```xml
-<PackageReference Include="Azure.Identity"
+<PackageReference Include="Aspire.Azure.Npgsql"
                   Version="*" />
 ```
 
 ---
 
-The PostgreSQL connection can be consumed using the client integration and <xref:Azure.Identity>:
+<!-- TODO: Add xref to AddAzureNpgsqlDataSource when available -->
+
+The PostgreSQL connection can be consumed using the client integration by calling the `AddAzureNpgsqlDataSource`:
+
+```csharp
+builder.AddAzureNpgsqlDataSource(connectionName: "postgresdb");
+```
+
+> [!TIP]
+> The `connectionName` parameter must match the name used when adding the PostgreSQL server resource in the app host project.
+
+The preceding code snippet demonstrates how to use the `AddAzureNpgsqlDataSource` method to register an `NpgsqlDataSource` instance that uses Azure authentication ([Microsoft Entra ID](/azure/postgresql/flexible-server/concepts-azure-ad-authentication)). This `"postgresdb"` connection name corresponds to a connection string configuration value.
+
+After adding `NpgsqlDataSource` to the builder, you can get the `NpgsqlDataSource` instance using dependency injection. For example, to retrieve your data source object from an example service define it as a constructor parameter and ensure the `ExampleService` class is registered with the dependency injection container:
+
+```csharp
+public class ExampleService(NpgsqlDataSource dataSource)
+{
+    // Use dataSource...
+}
+```
+
+For more information on dependency injection, see [.NET dependency injection](/dotnet/core/extensions/dependency-injection).
+
+### Add keyed Azure Npgsql client
+
+<!-- TODO: Add xref to AddKeyedAzureNpgsqlDataSource when available -->
+
+There might be situations where you want to register multiple `NpgsqlDataSource` instances with different connection names. To register keyed Npgsql clients, call the `AddKeyedAzureNpgsqlDataSource` method:
+
+```csharp
+builder.AddKeyedAzureNpgsqlDataSource(name: "chat");
+builder.AddKeyedAzureNpgsqlDataSource(name: "queue");
```

Comment:
"chat" and "queue" are a little weird for different datasource names.

Reply:
Hints: "website[db]", "store[db]", "crm[db]", because it's not that easy to find good examples

Reply:
Fixed